rubicon/objc/core_foundation.py

```diff
@@ -209,6 +216,18 @@ def from_value(value):
         return at(value.decode('utf-8'))
     elif isinstance(value, Decimal):
         return NSDecimalNumber.from_decimal(value)
+    elif isinstance(value, dict):
```

Comment:
Should we perhaps check for `collections.abc.Mapping` instead of `dict` here, so it works with other mapping types (`defaultdict`, `OrderedDict`, `Counter`, etc.) as well? On the other hand that would require an extra check so we don't "convert" `ObjCDictionaryInstance`s to Python and back again. Maybe at the top we should just add `if isinstance(value, ObjCInstance): return value`, so that Objective-C objects are never touched, no matter how much they pretend to be native Python objects.

Reply:
Turns out is isn't needed (at least for the examples you give):
```
>>> from collections import OrderedDict, defaultdict, Counter
>>> isinstance(OrderedDict(), dict)
True
>>> isinstance(defaultdict(), dict)
True
>>> isinstance(Counter(), dict)
True
```
They're all subclasses of the builtin dict, so they'll get converted automatically.

Reply:
Okay, `collections.ChainMap`. :P In any case, I would be fine with either option. Even if we require an actual `dict` or subclass, users can use an explicit `dict(non_dict_mapping)`.

Reply:
Exactly - plus, it's a little more explicit, which helps avoid nasty surprises when you have a dict-like class that isn't *quite* a dict.